### PR TITLE
drivers: video: mt9m114: configure camera with devicetree

### PR DESCRIPTION
**Pull request by DerekSnell**
Replace hard-coded configuration with devicetree properties.

### CODE REVIEW

File: dts/bindings/video/aptina,mt9m114.yaml
Comment (pdgendt) [inline dts/bindings/video/aptina,mt9m114.yaml:1]: All the properties have default values, unless these match some datasheet pre-defined configuration, these should be left out and add a `required: true` option. 
This "forces" the user to think of a value that matches the hardware.

**Review by gmarull — CHANGES_REQUESTED** [review]
binding defaults need justification, please, check the guidelines.

**Comment by ngphibang** [comment]
Hi Derek,

I agree that hardcoding the camera settings is not good but I don’t think exposing all the settings in the DT is a correct way to fix it. In a camera driver, there is usually an init params array containing default parameters. Then, if we want to change
•	some settings like 3A, contrast, etc., it should be done via the set_ctrl API. You could find an example of this in the ov2640 driver. I agree that VIDEO_CID_ controls could be not complete in Zephyr but if it lacks some controls ID, we could extend it.
•	settings like format / resolution should be changed via the get/set_format APIs. The video capture sample currently uses the default format/resolution set by the camera sensor driver at init time but an application should be able to get the capability of the camera pipeline and choose or set the format it wants. This "desired" format will be propagated from application through the camera pipeline to the camera sensor and could be changed when arrived at the sensor as elements along the pipeline will adapt the format based on their capability.

**Comment by loicpoulain** [comment]
Also, agree that devicetree is normally for hardware description, not configuration. Lot of configuration here.

**Comment by DerekSnell** [comment]
Thank you all for the feedback.  I will rework this PR.

**Comment by DerekSnell** [comment]
Closing, since this merged PR is a better solution https://github.com/zephyrproject-rtos/zephyr/pull/70237

**Pull request closed without merging**